### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by Koshao-sys**
Hi,

**@Odongo006  and @Koshao-sys we worked on the following requirements in this PR,**

- Create a file named schema_based_on_diagram.sql where you implement the database from the diagram.

  - Create Patients table
    create table patients

  - Create medical_histories table
    create table medical_histories

  - Create Invoices table
    create table invoices

  - Create treatments table
    create table treatments

  - Create invoice_items table
    create table invoice items

- Join tables from many-to-many relationships might not appear in the diagram, but you still need them.
  - Create join table for many-to-many relationship between medical_histories and treatments.
    create join table

- add the FK indexes as below:

  index on patient

  index on medical history

  index on invoice one

  index on invoise two

  index on join table one

  index on join table two

### CODE REVIEW

**Review by brainconnect93 — COMMENTED** [review]
Hi @Koshao-sys & @Odongo006,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

You have submitted a project that has the wrong workflow. According to the project requirement, it was stated that we should follow `Gitflow` in creating our branches. Kindly work on this. Thank You

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

_Cheers and Happy coding!👏👏👏_

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

-----

_Invalid Code Review Request does not count into the code reviews limit._

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+quantity int,
+total_price decimal,
+invoice_id int references invoices(id),
+treatment_id int references treatments(id));
+
+/* join table for many to many relationship between medical_histories and treatments */
+create table medical_histories_treatments(
+id int generated always as identity primary key,
+medical_history_id int references medical_histories(id),
+treatment_id int references treatments(id));
```
Comment (gemmen29) [inline schema_based_on_diagram.sql]: - [ ] Kindly create indexes for all foreign keys as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BregornOriginal — APPROVED** [review]
Hi @Koshao-sys 

Your project is complete!  There is nothing else to say other than... it's time to merge it 

Congratulations! 



## Highlights

✅ Good work implementing INDEX 💯 

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Comment by Koshao-sys** [comment]
@BregornOriginal Thanks for the review

**Pull request merged**